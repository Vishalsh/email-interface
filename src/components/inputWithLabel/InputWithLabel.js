import React from 'react';
import PropTypes from 'prop-types';

import classes from './InputWithLabel.module.scss';

const InputWithLabel = (props) => {
  const {
    label,
    value,
    inputType,
    textarea,
    onChange
  } = props;

  return (
    <div className={classes.inputWithLabel}>
      <label className={classes.label}
             htmlFor={label}>
        {label.toUpperCase()}
      </label>
      {
        textarea ?
          <textarea id={label}
                    className={`${classes.input} ${classes.textarea}`}
                    value={value}
                    name={label}
                    onChange={onChange}>
          </textarea>
          :
          <input id={label}
                 className={classes.input}
                 value={value}
                 name={label}
                 type={inputType}
                 onChange={onChange}/>
      }
    </div>
  );
};

InputWithLabel.defaultProps = {
  inputType: 'text',
  textarea: false
};

InputWithLabel.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  inputType: PropTypes.string,
  textarea: PropTypes.bool,
  onChange: PropTypes.func.isRequired
};

export default InputWithLabel;
