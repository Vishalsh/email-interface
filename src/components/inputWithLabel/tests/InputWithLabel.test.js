import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import InputWithLabel from '../InputWithLabel';

describe('InputWithLabel', () => {
  it('should render the component', () => {
    const props = {
      label: 'foo',
      value: 'bar',
      onChange: jest.fn()
    };
    const component = shallow(<InputWithLabel {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render the component with textarea', () => {
    const props = {
      label: 'foo',
      value: 'bar',
      textarea: true,
      onChange: jest.fn()
    };
    const component = shallow(<InputWithLabel {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  })
});