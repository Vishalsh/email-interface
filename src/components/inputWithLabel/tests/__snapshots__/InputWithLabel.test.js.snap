// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithLabel should render the component 1`] = `
<div
  className="inputWithLabel"
>
  <label
    className="label"
    htmlFor="foo"
  >
    FOO
  </label>
  <input
    className="input"
    id="foo"
    name="foo"
    onChange={[MockFunction]}
    type="text"
    value="bar"
  />
</div>
`;

exports[`InputWithLabel should render the component with textarea 1`] = `
<div
  className="inputWithLabel"
>
  <label
    className="label"
    htmlFor="foo"
  >
    FOO
  </label>
  <textarea
    className="input textarea"
    id="foo"
    name="foo"
    onChange={[MockFunction]}
    value="bar"
  />
</div>
`;
